src/runner/runESLint.js

```diff
@@ -96,8 +98,55 @@ const getComputedFixValue = ({ fix, quiet, fixDryRun }) => {
   return undefined;
 };
 
+const useFlatConfig = async () =>
+  !!(await findUp('eslint.config.js', { cwd: process.cwd() }));
```

Comment:
tbh i'm not sure this is sufficient, since `eslint -c eslint.config.js` would be the legacy config system in eslint 7.

Reply:
[I'm not sure if that's true](https://github.com/eslint/eslint/blob/559ff4e4bc54a8b6e6b54825d83c532d724204b3/lib/options.js#L153-L160) because reading through the option code it looks like `-c` only affects the name of the config file, regardless of the config system.  It looks like the flat config is selected in one of two cases:

1. [`eslint.config.js` exists in the tree](https://github.com/eslint/eslint/blob/559ff4e4bc54a8b6e6b54825d83c532d724204b3/lib/eslint/flat-eslint.js#L266-L271)
2. [`process.env.SHOULD_USE_ESLINT` is 'true'](https://github.com/eslint/eslint/blob/559ff4e4bc54a8b6e6b54825d83c532d724204b3/lib/cli.js#L283-L301)

I could add the case for the `SHOULD_USE_ESLINT` env var but I don't see in the eslint code where `-c` would use the old config format.  I'm no expert though so please correct me if I'm mistaken.

Reply:
I meant in eslint 7, which we still support.

Reply:
Ahh right, I can add a semver check in there too, do you think that's the right fix?

Reply:
That seems reasonable, assuming 8.0.0 is what has flat config in it.

Reply:
fixed, thanks